panic: runtime error: invalid memory address or nil pointer dereference when generating @pulumi/pulumi Node.js docs

```
Generating pulumi.com API docs
go run ./tools/tscdocgen/main.go ./tools/tscdocgen/template.go ../pulumi/sdk/nodejs pulumi /var/folders/fr/m76cr81124zd3kyjd2smrfnc0000gn/T/tmp.L18akBj9/pulumi.docs.json ./content/docs/reference/pkg/nodejs/pulumi/pulumi ./data/pkg/nodejs/pulumi pulumi/sdk/nodejs 57b6e84645cbaff8da8ab669c57a6e31f7cb989b
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x132b5a3]

goroutine 1 [running]:
main.(*emitter).createTypeLabel(0xc00028de88, 0x0, 0x1, 0xc0005d0000, 0x4)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:1058 +0x43
main.(*emitter).createTypeLabel(0xc00028de88, 0xc000429290, 0x0, 0xc0005bf950, 0x27)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:1110 +0xc71
main.(*emitter).createCodeDetails(0xc00028de88, 0xc00042ac40, 0xc00042a540, 0x13e550d)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:932 +0x332
main.(*emitter).augmentNode(0xc00028de88, 0xc00042ac40, 0xc00042a540, 0x0)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:269 +0x100
main.(*emitter).augmentNode(0xc00028de88, 0xc00042a540, 0xc00025e000, 0x0)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:311 +0x547
main.(*emitter).augmentNode(0xc00028de88, 0xc00025e000, 0x0, 0x0)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:311 +0x547
main.emitMarkdownDocs(0x7ffeefbff7b9, 0x14, 0x7ffeefbff7ce, 0x6, 0xc00025e000, 0x7ffeefbff824, 0x31, 0x7ffeefbff856, 0x18, 0x7ffeefbff86f, ...)
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:210 +0x1f0
main.main()
	/Users/justin/go/src/github.com/pulumi/docs/tools/tscdocgen/main.go:51 +0x2d2
exit status 2
```

This happens with `v1.0.0-beta.1` and more recent, but does not happen on `v0.17.28`.

--- a/tools/tscdocgen/main.go
+++ b/tools/tscdocgen/main.go
@@ -1092,8 +1092,13 @@ func (e *emitter) createTypeLabel(t *typeDocType, indent int) string {
 			label := "{\n"
 			indent++
 			for _, child := range decl.Children {
-				label += fmt.Sprintf("%s%s: %s;\n",
-					strings.Repeat(" ", indent*4), child.Name, e.createTypeLabel(child.Type, indent))
+				if child.Kind == typeDocFunctionNode && len(child.Signatures) > 0 {
+					label += fmt.Sprintf("%s%s;\n",
+						strings.Repeat(" ", indent*4), e.createSignature(child.Signatures[0], decl, false))
+				} else {
+					label += fmt.Sprintf("%s%s: %s;\n",
+						strings.Repeat(" ", indent*4), child.Name, e.createTypeLabel(child.Type, indent))
+				}
 			}
 			indent--
 			return fmt.Sprintf("%s%s}", label, strings.Repeat(" ", indent*4))

